java-checks/src/main/java/org/sonar/java/checks/ChildClassShadowFieldCheck.java

```diff
@@ -65,26 +65,29 @@ public void visitNode(Tree tree) {
         if (member.is(Tree.Kind.VARIABLE)) {
           String fieldName = ((VariableTree) member).simpleName().name();
           if (!IGNORED_FIELDS.contains(fieldName)) {
-            Symbol.TypeSymbol definingClass = getDefiningClassInHierarchyForField(superclassSymbol, fieldName);
-            if (definingClass != null) {
-              addIssue(member, String.format("\"%s\" is the name of a field in \"%s\".", fieldName, definingClass.name()));
-            }
+            checkForIssue(superclassSymbol, member, fieldName);
           }
         }
       }
     }
   }
 
-  @CheckForNull
-  private Symbol.TypeSymbol getDefiningClassInHierarchyForField(Symbol.TypeSymbol classSymbol, String fieldName) {
+  private void checkForIssue(Symbol.TypeSymbol classSymbol, Tree memberTree, String fieldName) {
+    String upperCaseFieldName = fieldName.toUpperCase();
```

Comment:
This should probably be uppercased directly in the condition that tests it, not prior to this.